app/src/main/java/space/taran/arknavigator/mvp/model/repo/stats/category/StatsCategoryStorage.kt

```diff
@@ -0,0 +1,40 @@
+package space.taran.arknavigator.mvp.model.repo.stats.category
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.debounce
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.launch
+import space.taran.arknavigator.mvp.model.arkFolder
+import space.taran.arknavigator.mvp.model.arkStats
+import space.taran.arknavigator.mvp.model.repo.stats.StatsCategory
+import space.taran.arknavigator.mvp.model.repo.stats.StatsEvent
+import java.nio.file.Path
+
+private const val DEBOUNCE_FLUSH = 30_000L
```

Comment:
Would flushing be triggered if user performs some action fitting to some category, then immediately switching to other app? In other words, does debouncing timer works when the app is minimized?

Reply:
Also, maybe `DEBOUNCE_FLUSH_INTERVAL` or just `FLUSH_INTERVAL` would be a better name.

Reply:
Also, preference in Settings screen might be handy. Not for users, but for a developer. I suspect, lowering this interval down to 5 seconds should be still fine.

Reply:
> Would flushing be triggered if user performs some action fitting to some category, then immediately switching to other app? In other words, does debouncing timer works when the app is minimized?

Yes, timer works when app is minimized

> Also, maybe DEBOUNCE_FLUSH_INTERVAL or just FLUSH_INTERVAL would be a better name.

`FLUSH_INTERVAL` sounds good

> Also, preference in Settings screen might be handy. Not for users, but for a developer. I suspect, lowering this interval down to 5 seconds should be still fine.

Sure